### ISSUE TITLE
Make SET command use the appropriate datatype when saving data in store

### ISSUE THREAD

**Issue by yashs360** [DiceDB/dice#359]
Currently the SET command is using `string` data type for all commands, even when a passed command uses numeric values. 

Steps to reproduce - 

```
QWATCH "SELECT $key, $value FROM `match:100:*` WHERE $value > 10 ORDER BY $value DESC LIMIT 3"
(empty list or set)
127.0.0.1:7379>
127.0.0.1:7379>
127.0.0.1:7379> SET match:100:user:1 1
OK
127.0.0.1:7379>
```

Error - 

```
2024/08/20 13:52:50 ERRO incompatible types in comparison: string and int

```

<img width="772" alt="SET-datatype" src="https://github.com/user-attachments/assets/c660db2a-db57-401b-8444-136b73fc1732">

**Comment by gauravsarma1992** [comment]
@JyotinderSingh  I will take this up.

**Comment by JyotinderSingh** [comment]
Added fix in #363.
